pkg/controller/clusterautoscaler/clusterautoscaler_controller.go

```diff
@@ -143,10 +144,15 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		return reconcile.Result{}, err
 	}
 
+	// caRef is a reference to the ClusterAutoscaler object, but with the
+	// namespace for cluster-autoscaler deployments set.  This keeps events
+	// generated for these cluster scoped objects out of the default namespace.
+	caRef := r.objectReference(ca)
```

Comment:
Can `caRef` ever be `nil`?

Reply:
In practice it won't here, but yes it can. It will just cause subsequent event recording to fail and log a warning in this reconciliation. It won't affect anything else.

Reply:
How can it be CAO expects CA to be under `request.NamespacedName`. Yet, when you get the CA object from the namespace, the namespace is set to `""`?

Reply:
Exactly, so if you pass that ClusterAutoscaler object with an empty namespace to an even recorder, it places the event (which are namespaced objects) into the default namespace. This avoids that by creating an ObjectReference  pointing to the ClusterAutoscaler but explicitly adding a namespace. That reference is only used by the event recorder.

The alternative is to give this operator permissions to manage events in the default namespace, and to not have the events show up if you list all events in the `openshift-machine-api` namespace.

Reply:
Sorry, maybe I misunderstood. The namespace field of the `NamespacedName` here is *always* empty because ClusterAutoscaler resources are cluster-scoped. They don't have an associated namespace, just like Node or ClusterOperator objects for example.